sympy/core/mul.py

```diff
@@ -1245,7 +1245,7 @@ def as_base_exp(self):
                 nc += 1
             if e1 is None:
                 e1 = e
-            elif e != e1 or nc > 1:
+            elif e != e1 or nc > 1 or not e.is_Integer:
```

Comment:
should that be `e.is_integer` to allow for `Symbol('e', integer=True)`?

Reply:
No, we should not invoke the assumptions system in such a basic manipulation. I would rather remove this whole method but the change here is intended to be a minimal change that fixes the correctness problem.